### PR TITLE
fix: relax assertion into error

### PR DESCRIPTION
**Pull request by ronag**
Relaxes assertion into an error which can be caught. Also adds missing return -1.

Refs: https://github.com/nodejs/undici/issues/756

### CODE REVIEW

**Comment by ronag** [comment]
Trying to resolve https://github.com/nodejs/undici/issues/756. Currently just guesswork since we can't reproduce the issue.

File: deps/llhttp/src/api.c [hunk 1]
```diff
@@ -41,6 +41,7 @@ void llhttp_init(llhttp_t* parser, llhttp_type_t type,
 
 #if defined(__wasm__)
 
+extern int wasm_on_message_begin(llhttp_t* p);
```
Comment (mcollina) [inline deps/llhttp/src/api.c:44]: have you added this or is it part of llhttp?

**Review by mcollina — COMMENTED** [review]
good spot!

File: deps/llhttp/src/api.c (hunk 1 above)
Comment (ronag) [reply deps/llhttp/src/api.c:44]: It's part of llhttp.

File: deps/llhttp/src/api.c [hunk 2]
```diff
@@ -53,7 +54,7 @@ int on_headers_complete(llhttp_t * p)
 }
 
 const llhttp_settings_t wasm_settings = {
-  NULL,
+  wasm_on_message_begin,
```
Comment (ronag) [inline deps/llhttp/src/api.c:57]: @mcollina previously we didn't use this llhttp callback and just passed NULL.

Comment (mcollina) [reply deps/llhttp/src/api.c:57]: I mean.. this change shouldn't happen upstream?

Comment (ronag) [reply deps/llhttp/src/api.c:57]: We have other changes here that are not upstream.

Comment (ronag) [reply deps/llhttp/src/api.c:57]: I'm not sure it's worth trying to port these changes upstream as it's just C glue between undici and llhttp.

**Review by dnlup — APPROVED** [review]

File: deps/llhttp/src/api.c (hunk 2 above)
Comment (mcollina) [reply deps/llhttp/src/api.c:57]: We should move them upstream as it will be simpler to bump the library.

**Review by mcollina — APPROVED** [review]
lgtm

File: deps/llhttp/src/api.c (hunk 2 above)
Comment (ronag) [reply deps/llhttp/src/api.c:57]: https://github.com/nodejs/llhttp/pull/116

**Pull request merged**